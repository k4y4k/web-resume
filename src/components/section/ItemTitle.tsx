import * as React from 'react'
import tw, { TwStyle } from 'twin.macro'

interface ItemTitleTypes {
  title: string | undefined
}

const itemTitleStyles = (): Array<TwStyle | string> => [tw`text-2xl uppercase`]

const ItemTitle = ({ title }: ItemTitleTypes): JSX.Element => {
  if (title === '' || title === undefined)
    return <h1 css={itemTitleStyles()}>Error: No Item Title</h1>

  return <h1 css={itemTitleStyles()}>{title}</h1>
}

export default ItemTitle
