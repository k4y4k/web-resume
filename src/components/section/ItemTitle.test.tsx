import * as React from "react";
import { render, screen } from "@testing-library/react";
import ItemTitle from "./ItemTitle";

describe("<ItemTitle />", () => {
  test("handles no data", () => {
    render(<ItemTitle title="" />);

    const title = screen.queryByTestId("itemTitle");
    expect(title).not.toBeInTheDocument();
    expect(title).toMatchSnapshot();
  });

  test("displays title", () => {
    render(<ItemTitle title="My Frankly Astounding Thing" />);

    const title = screen.getByTestId("itemTitle");
    expect(title).toHaveTextContent(/my frankly astounding thing/i);
    expect(title).toMatchSnapshot();
  });
});
