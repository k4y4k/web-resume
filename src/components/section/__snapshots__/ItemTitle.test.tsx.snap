// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemTitle /> displays title 1`] = `
.c0 {
  margin-bottom: 0.25rem;
  font-weight: 700;
  font-size: 1.25rem;
  line-height: 1.25rem;
}

<h1
  class="c0"
  data-testid="itemTitle"
>
  My Frankly Astounding Thing
</h1>
`;

exports[`<ItemTitle /> handles no data 1`] = `null`;
